### PR TITLE
[DD] New: Half Adder post

### PR DESCRIPTION
**Pull request by kyaso**
Info: This post introduces the reader to the basic principles behind addition in binary plus the half adder.

Post checklist:

- [x] YAML: permalink
- [x] YAML: excerpt
- [x] YAML: description
- [x] YAML: author
- [x] YAML: toc label
- [x] YAML: nav sidebar
- [x] Ran Grammarly?
- [x] Added some images?
- [x] SEO: keywords in title, description, first 100 words
- [x] image have alt and title att?
- [x] no h1 header except title
- [x] Ran local test?
- [x] All links work?
- [x] Date correct?
- [x] Squashed commits?

### CODE REVIEW

File: _posts/2023-03-22-dd_00005_half_adder.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+| 1 | 0 | 1   | 0     |
+| 1 | 1 | 1   | 1     |
+
+
+Do you notice something? Right, this looks like a logic gate **truth table**! And that's precisely how we can come up with a logic circuit for adding two 1-bit numbers. This circuit has two inputs, the two numbers A and B, and two outputs, the sum and the carry. The corresponding circuit looks as follows:
+
+==TODO: HA circuit==
+
+This circuit looks pretty simple, just one XOR gate whose output is the sum bit and an AND gate where the carry comes from. If you look at the truth table above, and remember how the truth tables of the AND and XOR gate look like, you can easily see why we need these two gates. This circuit is called **Half-Adder**.
+
```
Comment (kyaso) [inline _posts/2023-03-22-dd_00005_half_adder.md]: Also add Half adder gate picture here

File: _posts/2023-03-22-dd_00005_half_adder.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+  digital addition circuitry."
+author: Kirthihan Yasotharan
+toc: false
+toc_label: "Content"
+toc_sticky: true
+toc_icon: "align-justify"
+header:
+  overlay_color: "#000"
+  overlay_filter: "0.5"
+  overlay_image: /assets/images/digital_design.jpg
```
Comment (leboem) [inline _posts/2023-03-22-dd_00005_half_adder.md]:
```suggestion
  overlay_image: /assets/images/digital_design.jpeg
```

File: _posts/2023-03-22-dd_00005_half_adder.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+---
+
+In our last two posts we have learned what logic gates are, which kind of gates
+exist and how they can be implemented using transistors.
+
+Now it's time to see what all stuff we can do with these things! Have you ever
```
Comment (leboem) [inline _posts/2023-03-22-dd_00005_half_adder.md]:
```suggestion
Now it's time to see what stuff we can do with these things! Have you ever
```

File: _posts/2023-03-22-dd_00005_half_adder.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+wondered why a computer is called "computer"? Well, as the name suggests,
+computers calculate, or _compute_, stuff. What and how they do it is something
```
Comment (leboem) [inline _posts/2023-03-22-dd_00005_half_adder.md]: maybe not again the word "stuff"

File: _posts/2023-03-22-dd_00005_half_adder.md [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+we will cover in the future, but for now we'll focus on the essentials.
+
+It turns out that we can use logic gates to do all kinds of **mathematical
+operations**. In this post, we will have a look at the most basic of
+mathematical operations: the **addition**. How can we add two numbers using just
```
Comment (leboem) [inline _posts/2023-03-22-dd_00005_half_adder.md]:
```suggestion
mathematical operation: the **addition**. How can we add two numbers using just
```

File: _posts/2023-03-22-dd_00005_half_adder.md [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+hardware side, we assume that you are already familiar with **binary numbers**.
+If not, or you need to refresh your memory, then feel free to consult a search
+engine and/or latest AI of your choice to learn the basics about binary numbers.
+We recommend giving the
+[Wikipedia article](https://en.wikipedia.org/wiki/Binary_number) about binary
+numbers a read. Of course, there are also plenty of tutorials on YouTube, such
+as [this](https://www.youtube.com/watch?v=LpuPe81bc2w) one on the basics of
+binary numbers, followed by [this](https://www.youtube.com/watch?v=C5EkxfNEMjE)
+one on doing addition in the binary system. We might also cover binary numbers
+in a future post in our _Getting Started_ category, so stay tuned for that. Ok,
```
Comment (leboem) [inline _posts/2023-03-22-dd_00005_half_adder.md]: maybe a link to getting started category?

**Review by leboem — CHANGES_REQUESTED** [review]
Very nice article :-)

File: _posts/2023-03-22-dd_00005_half_adder.md (hunk 2 above)
Comment (kyaso) [reply _posts/2023-03-22-dd_00005_half_adder.md]: Good catch!

Fixed [f3164cf](https://github.com/bitsandvolts/bitsandvolts.github.io/pull/49/commits/f3164cf2995680c881a9fb281171cc1d8b51049a)

File: _posts/2023-03-22-dd_00005_half_adder.md (hunk 3 above)
Comment (kyaso) [reply _posts/2023-03-22-dd_00005_half_adder.md]: Fixed: [db38042](https://github.com/bitsandvolts/bitsandvolts.github.io/pull/49/commits/db3804213df03bf107a353592d84004def502a16)

File: _posts/2023-03-22-dd_00005_half_adder.md (hunk 4 above)
Comment (kyaso) [reply _posts/2023-03-22-dd_00005_half_adder.md]: Fixed: [9c71535](https://github.com/bitsandvolts/bitsandvolts.github.io/pull/49/commits/9c71535a9711473a1972f814625dde3ef1fc7af2)

Is "things" better?

File: _posts/2023-03-22-dd_00005_half_adder.md (hunk 5 above)
Comment (kyaso) [reply _posts/2023-03-22-dd_00005_half_adder.md]: Fixed: [8d0a5ac](https://github.com/bitsandvolts/bitsandvolts.github.io/pull/49/commits/8d0a5aca70a16a107a0ccf626a434b1d8934e5c2)

File: _posts/2023-03-22-dd_00005_half_adder.md (hunk 6 above)
Comment (kyaso) [reply _posts/2023-03-22-dd_00005_half_adder.md]: Good point -> fixed: [a48611c](https://github.com/bitsandvolts/bitsandvolts.github.io/pull/49/commits/a48611c7f41043e744373327dc0e13fdf157dbe0)

File: _posts/2023-03-22-dd_00005_half_adder.md [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+header:
+  overlay_color: "#000"
+  overlay_filter: "0.5"
+  overlay_image: /assets/images/digital_design.jpeg
+---
+
+In our last two posts we have learned what logic gates are, which kind of gates
+exist and how they can be implemented using transistors.
+
+Now it's time to see what stuff we can do with these things! Have you ever
```
Comment (leboem) [inline _posts/2023-03-22-dd_00005_half_adder.md]:
```suggestion
Now it's time to see what we can do with these things! Have you ever
```

Comment (leboem) [reply _posts/2023-03-22-dd_00005_half_adder.md]: Then you can use stuff below

File: _posts/2023-03-22-dd_00005_half_adder.md [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+wondered why a computer is called "computer"? Well, as the name suggests,
+computers calculate, or _compute_, things. What and how they do it is something
```
Comment (leboem) [inline _posts/2023-03-22-dd_00005_half_adder.md]:
```suggestion
computers calculate, or _compute_, stuff. What and how they do it is something
```

File: _posts/2023-03-22-dd_00005_half_adder.md [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+we will cover in the future, but for now we'll focus on the essentials.
+
+It turns out that we can use logic gates to do all kinds of **mathematical
+operations**. In this post, we will have a look at the most basic mathematical
+operation: the **addition**. How can we add two numbers using just simple logic
+gates, you ask?
+
+Well, for that we need some basic understanding of **binary numbers** and how
+addition works in the binary system. As this blog series focuses more on the
+hardware side, we assume that you are already familiar with **binary numbers**.
```
Comment (leboem) [inline _posts/2023-03-22-dd_00005_half_adder.md]:
```suggestion
hardware side, we assume that you are already familiar with binary numbers.
```

Comment (leboem) [reply _posts/2023-03-22-dd_00005_half_adder.md]: remember: no duplicate bold

File: _posts/2023-03-22-dd_00005_half_adder.md [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+This is yet another example for <i>abstraction</i>: We simplify things by hiding the
+lower-level details of a particular system. In our case, the low-level details
+were the AND and XOR gates of the half-adder. And remember, those gates
+themselves again were an abstraction of the underlying transisitor circuits!
+</div>
+
+You might think that now extending it to adding larger than 1-bit numbers is as
+simple as taking a bunch of half-adders for each digit, right? Well, almost. We
+need to extend our circuit a bit. Why? Because we also need to consider the
+_carry_ bit coming from the _previous_ digits (except the first digits). The
```
Comment (leboem) [inline _posts/2023-03-22-dd_00005_half_adder.md]:
```suggestion
_carry_ bit coming from the _previous_ digits (except for the first digits). The
```

Comment (leboem) [reply _posts/2023-03-22-dd_00005_half_adder.md]: Otherwise, it sounds a bit like we ignore the carry coming from the first digits

**Review by leboem — CHANGES_REQUESTED** [review]
Two more things:

- We usually have a summary at the end. I agree that it is not really required here. Up to you ...
- The images are very bright. In your past posts you used dark background

File: _posts/2023-03-22-dd_00005_half_adder.md (hunk 4 above)
Comment (leboem) [reply _posts/2023-03-22-dd_00005_half_adder.md]: see my suggestions below

**Review by leboem — DISMISSED** [review]
LGTM

**Review by leboem — APPROVED** [review]
LGTM

**Pull request merged**